tests/golos.publication_tests.cpp

```diff
@@ -131,27 +141,60 @@ class golos_publication_tester : public golos_tester {
         ("jsonmetadata", "jsonmetadata");
 
     struct errors: contract_error_messages {
-        const string msg_exists         = amsg("This message already exists.");
-        const string unregistered_user_ = amsg("unregistered user: ");
-        const string no_content         = amsg("Content doesn't exist.");
-
-        const string delete_children    = amsg("You can't delete comment with child comments.");
-        const string no_message         = amsg("Message doesn't exist.");
-        const string vote_same_weight   = amsg("Vote with the same weight has already existed.");
-        const string vote_weight_0      = amsg("weight can't be 0.");
-        const string vote_weight_gt100  = amsg("weight can't be more than 100%.");
-        const string no_revote          = amsg("You can't revote anymore.");
-        const string upvote_near_close  = amsg("You can't upvote, because publication will be closed soon.");
-        const string max_comment_depth  = amsg("publication::create_message: level > MAX_COMMENT_DEPTH");
+        const string msg_exists            = amsg("This message already exists.");
+        const string unregistered_user_    = amsg("unregistered user: ");
+        const string no_content            = amsg("Content doesn't exist.");
+
+        const string delete_children       = amsg("You can't delete comment with child comments.");
+        const string no_message            = amsg("Message doesn't exist.");
+        const string vote_same_weight      = amsg("Vote with the same weight has already existed.");
+        const string vote_weight_0         = amsg("weight can't be 0.");
+        const string vote_weight_gt100     = amsg("weight can't be more than 100%.");
+        const string no_revote             = amsg("You can't revote anymore.");
+        const string upvote_near_close     = amsg("You can't upvote, because publication will be closed soon.");
+        const string max_comment_depth     = amsg("publication::create_message: level > MAX_COMMENT_DEPTH");
+        const string window_less_0         = amsg("Cashout window must be greater than 0.");
+        const string wndw_less_upvt_lckt   = amsg("Cashout window can't be less than upvote lockout.");
+        const string max_cmmnt_dpth_less_0 = amsg("Max comment depth must be greater than 0.");
     } err;
 };
 
 
 BOOST_AUTO_TEST_SUITE(golos_publication_tests)
 
+BOOST_FIXTURE_TEST_CASE(set_params, golos_publication_tester) try {
+    BOOST_TEST_MESSAGE("Test posting parameters");
+    
+    produce_block();
+    
+    BOOST_TEST_MESSAGE("--- check that global params not exist");
+    BOOST_TEST_CHECK(post.get_params().is_null());
+
+    init_params();
+
+    auto obj_params = post.get_params();
+    BOOST_TEST_MESSAGE("--- " + fc::json::to_string(obj_params));
```

Comment:
this output was for debug purposes, but now it copied to almost every params test : )